### PR TITLE
[#102] Automate New version bumping by Github action

### PR DESCRIPTION
**Pull request by doannimble**
https://github.com/nimblehq/flutter_templates/issues/102

## What happened 👀

Create a github action to bump version automatically

## Insight 📝

Automate the process for new pull request creating to bump new version after each release:
- Checkout.
- Bump the new version.
- Commit, push & create a new pull request for new version bumping.

## Proof Of Work 📹

https://github.com/nimblehq/flutter_templates/pull/126/files

https://user-images.githubusercontent.com/88996251/209756853-68590462-b491-480c-ae5b-5415524f2c2b.mov

### CODE REVIEW

**Comment by luongvo** [comment]
@doannimble I just switched `default branch` to this branch for experiment 🙏

**Comment by luongvo** [comment]
@doannimble Thank you for contributing to this feature of the template 💪 

Could you please rebase to have around 2 commits: `[#102] Create bump_version.yml` and `[#102] Add bump version script`? There are too much commits in this PR 🙏

File: .github/workflows/bump_version.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the latest code
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set new version
+        id: set_version
+        run: |
+          perl -i -pe 's/^(version:\s+\d+\.\d+\.\d+\+)(\d+)$/"version: ${{ github.event.inputs.newVersion }}+".($2+1)/e' ./.template/pubspec.yaml
```
[the remark this answers is not in the document]
Comment (luongvo) [reply .github/workflows/bump_version.yml:15]: I think we should simply bump the new version from the Workflow input instead https://github.com/nimblehq/elixir-templates/blob/develop/.github/workflows/bump_version.yml#L3-L9

File: .github/workflows/bump_version.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+            Bump version to ${{ github.event.inputs.newVersion }}
+
+            ## Insight
+
+            Automatically created by the Bump Version workflow.
+
+            ## Proof Of Work
+
+            On the Files changed tab
```
[the remark this answers is not in the document]
Comment (doannimble) [reply .github/workflows/bump_version.yml:39]: At least `On the Files changed tab` having more meaning than `N/A` @luongvo  😄

File: .github/workflows/bump_version.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (doannimble) [reply .github/workflows/bump_version.yml:15]: @luongvo Yes, it's precisely what I'm doing now. The example [PR](https://github.com/nimblehq/flutter_templates/pull/126) is the result after inputting `1.3.1` , the versionCode with work for Android and I think that make sense to auto-increase it🙏

**Comment by luongvo** [comment]
@doannimble Oops, I missed your updates in this PR. Could you please make use of `Re-request review` feature later?

image

**Review by luongvo — COMMENTED** [review]
The rest lgtm. Can we have a POW video that shows how can we bump the new version PR by this workflow & how the new PR looks like after bumping?

File: .github/workflows/bump_version.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+name: Bump Version
+
+jobs:
```
Comment (luongvo) [inline .github/workflows/bump_version.yml:3]: I think we're missing this 
https://github.com/nimblehq/elixir-templates/blob/0f30d704740276815f54c4b4488d4dece7934141/.github/workflows/bump_version.yml#L3-L9

**Review by luongvo — APPROVED** [review]
lgtm

**Comment by doannimble** [comment]
@luongvo @manh-t @markgravity POW video added 🙏

**Review by manh-t — APPROVED** [review]

**Review by markgravity — APPROVED** [review]
Thank you for this great contribution 🥰

**Pull request merged**